docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,409 @@
+
+#### Usage
+
+The Layer Model component takes one prop:
+
+- `initialLayer` - defaults to `Content`, can be one of `Content`, `Build`, `Data`, `View`, `App` that correspond to the different layers
+
+```markdown
+---
+title: GraphQL Concepts in Gatsby
+---
+
+import LayerModel from "../../www/src/components/layer-model"
+
+To help understand how GraphQL works in Gatsby...
+
+<LayerModel initialLayer="Data" />
+```
+
+#### Sample
+
+When used, it looks like this:
+
+<LayerModel initialLayer="Data" />
+
+### Horizontal Navigation List
+
+The `<HorizontalNavList />` was made for the [Glossary](/docs/glossary/), and renders a list of links to alphabetical subheadings on the page in a horizontal format. Because of its specific use case, it isn't made globally available but can be imported if needed on other pages.
+
+#### Usage
+
+The Horizontal Nav List component takes two props:
+
+- `slug` - which is provided in the props of the page by default
+- `items` - an array of strings for items to render and wrap with a `<Link />` to subheadings
+
+<!-- prettier-ignore -->
+```markdown
+---
+title: Glossary
+---
+
+import HorizontalNavList from "../../www/src/components/horizontal-nav-list.js"
+
+The glossary defines key vocabulary...
+
+---
+
+<HorizontalNavList
+  slug={props.slug}
+  items={["guide-list", "egghead-embed", "pull-quote", "layer-model", "horizontal-navigation-list"]}
+/>
+```
+
+#### Sample
+
+Rendered, it looks like this:
+
+<HorizontalNavList
+items={[
+"guide-list",
+"egghead-embed",
+"pull-quote",
+"layer-model",
+"horizontal-navigation-list",
+]}
+slug={props.slug}
+/>
+
+### Breadcrumb
+
+The `<Breadcrumb />` component is used in layout files to display the hierarchy of pages a user is currently browsing on.
+
+#### Usage
+
+The Breadcrumb component takes one prop:
+
+- `location` - an object provided in the props of page templates by default
+- `itemList` - an object comprised of the docs hierarchical structure
+
+<!-- prettier-ignore -->
+```javascript
+import Breadcrumb from "../../www/src/components/docs-breadcrumb"
+
+<Breadcrumb location={props.location} itemList={itemList} />
+```
+
+To alter the title of a doc that is displayed in the Breadcrumb component, a `breadcrumbTitle` is supported as a key in the [sidebar YAML files](https://github.com/gatsbyjs/gatsby/tree/master/www/src/data/sidebars). It is commonly used to provide an abbreviated version of a doc's title, e.g. shortening "Adding a Custom webpack Config" to "webpack Config".
+
+#### Sample
+
+Rendered, it looks like this:
+
+<Breadcrumb
+location={{
+    pathname: "/contributing/docs-and-website-components/",
+  }}
+itemList={itemListContributing}
+/>
+
+---
+
+### Table of Contents
+
+The `<TableOfContents />` component is used to render a list of subheaders from a docs page and automatically provide deep links to them.
```

Comment:
Same as the breadcrumb: I think a link to an example in the source would be helpful.